sys/vane/jael.hoon

```diff
@@ -644,35 +647,103 @@
     ::
+    ::  boot from keys
+    ::    {$dawn =seed spon=(unit ship) czar=(map ship [=life =pass]) turf=(list turf)}
+    ::
+        %dawn
+      =*  our  who.seed.tac
+      ::  our initial private key, as a +tree of +rite
+      ::
+      =/  rit  (sy [%jewel (my [lyf.seed.tac key.seed.tac] ~)] ~)
+      =.  +>.$  $(tac [%mint our our rit])
+      ::  our initial galaxy table as a +map from +life to +public
+      ::
+      =/  kyz
+        %-  ~(run by czar.tac)
+        |=([=life =pass] `public`[live=| life (my [life pass] ~)])
+      =.  +>.$
+        %-  curd  =<  abet
+        (pubs:~(feel su hen our urb sub etn) kyz)
+      ::  XX sponsor
+      ::  XX reconcile with .dns.eth
+      ::  set initial domains
+      ::
+      =.  tuf.urb  turf.tac
+      ::
+      =.  moz
+        %+  weld  moz
+        ::  order is crucial!
+        ::
+        ::    %dill must init after %gall (and after %give intended for unix)
+        ::    %jael init must be deferred post %eyre init
+        ::
+        ^-  (list move)
+        :~  [hen %pass /(scot %p our)/init %b %wait +(now.sys)]
+            [hen %slip %d %init our]
+            [hen %give %init our]
+            [hen %slip %g %init our]
+            [hen %slip %e %init our]
+            [hen %slip %c %init our]
+            [hen %slip %a %init our]
+        ==
+      +>.$
+    ::
+    ::  boot fake
+    ::    {$fake our/ship}
+    ::
+        %fake
+      =*  our  our.tac
+      ::  our private key, as a +tree of +rite
+      ::
+      ::    Private key updates are disallowed for fake ships,
+      ::    so we do this first.
+      ::
+      =/  cub  (pit:nu:crub:crypto 512 our)
+      =/  rit  (sy [%jewel (my [1 sec:ex:cub] ~)] ~)
+      =.  +>.$  $(tac [%mint our our rit])
+      ::  set the fake bit
+      ::
+      =.  fak.urb  &
+      ::  initialize other vanes per the usual procedure
+      ::
+      ::    Except for ourselves!
+      ::
+      =.  moz
+        %+  weld  moz
+        ^-  (list move)
+        :~  [hen %slip %d %init our]
+            [hen %give %init our]
+            [hen %slip %g %init our]
+            [hen %slip %e %init our]
+            [hen %slip %c %init our]
+            [hen %slip %a %init our]
+        ==
+      +>.$
+    ::
     ::  remote update
     ::    {$hail p/ship q/remote}
     ::
         $hail
-      %+  cure  our.tac
+      %^  cure  hen  our.tac
       abet:abet:(hail:(burb our.tac) p.tac q.tac)
     ::
-    ::  initialize vane
-    ::    [%init our=ship]
-    ::
-        $init
-      %+  cute  our.tac  =<  abet
-      (~(init et our.tac now.sys etn.lex) our.tac)
-    ::
     ::  set ethereum source
     ::    [%look p=(each ship purl)]
     ::
         %look
-      %+  cute  our.tac  =<  abet
+      %^  cute  hen  our.tac  =<  abet
       (~(look et our.tac now.sys etn.lex) src.tac)
     ::
     ::  create promises
     ::    {$mint p/ship q/safe}
     ::
         $mint
-      %+  cure  our.tac
+      ~|  %fake-jael
+      ?<  fak.urb
```

Comment:
Wait, why wouldn't you be able to do non-PKI operations?

Reply:
We don't want to let fake ships rekey. This is a heavy handed way to do that. It'll blow up if we ever start using %jael for "promises". Which we aren't doing.

Reply:
> It'll blow up if we ever start using %jael for "promises".

My point. Can't we just ban `%jewel`s or whatever and call it a day?

Also, to be sure: we don't want fake ships to rekey because they technically only have one key (generated from shipnum), correct? What if we just incremented the key seed (ie shipnum) for every new life? Maybe too much work to be worth it.